DataDog/datadog-lambda-python
# Cover string/b64 msg attribute trace extraction

### What does this PR do?
Ports [Datadog/datadog-lambda-js#269](https://github.com/DataDog/datadog-lambda-js/pull/269) to Python Lambda Library and Tracer.


### Motivation
Fixes[Datadog/serverless-plugin-datadog#232](https://github.com/DataDog/serverless-plugin-datadog/issues/232) for Python lambda library and tracer. Message attributes sent as json strings break filter policies and make it so messages don't make it to subscribers with a filter policy.


### Testing Guidelines
Linking my local versions of the tracer and lambda library to my sample app I had a lambda function that published to an SNS topic that has an SQS subscription with a filter policy. After b64 encoding only SQS messageAttribute trace data in ddtrace-py the trace context is successfully passed through subscribers with a filter policy. I left SNS messageAttributes as strings. Check out my [DD TRACE PY PR](https://github.com/DataDog/dd-trace-py/pull/3404).

## Publisher Lambda->SNS->SQS-> Receiver Lambda ([dd link](https://app.datadoghq.com/functions?search=functionname%3Asns-filtersarod-dev-recieveletter%20functionname%3Asns-filtersarod-dev-lambdasnssqsreceivelambda&cloud=aws&graphType=flamegraph&panel_end=1647027023733&panel_paused=false&panel_start=1647023423733&panel_tab=invocations&selection=aws-lambda-functions%2Bsns-filtersarod-dev-lambdasnssqsreceivelambda%2Bsa-east-1%2B601427279990&spanID=15223171767780789530&traceID=480579719862754878&start=1647012561316&end=1647026961316&paused=false))
<img width="721" alt="lambdaSnsSqsLambda" src="https://user-images.githubusercontent.com/49878080/157968332-49d801a3-699a-4568-b80d-d9fcc2ae4b2e.png">


I also manually tested 

## Publisher Lambda->SNS-> Receiver Lambda [dd link](https://app.datadoghq.com/functions?search=functionname%3Asns-filtersarod-dev-recieveletter%20functionname%3Asns-filtersarod-dev-lambdasnssqsreceivelambda%20functionname%3Asns-filtersarod-dev-lambdasqsreceivelambda%20functionname%3Asns-filtersarod-dev-receiveletter&cloud=aws&graphType=flamegraph&panel_end=1647028793801&panel_paused=true&panel_start=1647014393801&panel_tab=invocations&selection=aws-lambda-functions%2Bsns-filtersarod-dev-receiveletter%2Bsa-east-1%2B601427279990&spanID=2776194193000583056&traceID=8219518836272491046&start=1647014393801&end=1647028793801&paused=false)

<img width="721" alt="lambdaSnsLambda" src="https://user-images.githubusercontent.com/49878080/157967944-f9653ddd-b5a7-4c1b-8682-f79a5993ad24.png">



## Publisher Lambda->SQS-> Receiver Lambda [dd link](https://app.datadoghq.com/functions?search=functionname%3Asns-filtersarod-dev-recieveletter%20functionname%3Asns-filtersarod-dev-lambdasnssqsreceivelambda%20functionname%3Asns-filtersarod-dev-lambdasqsreceivelambda%20functionname%3Asns-filtersarod-dev-receiveletter&cloud=aws&graphType=service_map&panel_end=1647032165650&panel_paused=false&panel_start=1647028565650&panel_tab=invocations&selection=aws-lambda-functions%2Bsns-filtersarod-dev-lambdasqsreceivelambda%2Bsa-east-1%2B601427279990&shouldShowLegend=false&spanID=12599013373589155477&spanViewType=logs&traceID=12535114939933676171&start=1647017523939&end=1647031923939&paused=false)

<img width="721" alt="Screen Shot 2022-03-11 at 4 07 03 PM" src="https://user-images.githubusercontent.com/49878080/157967799-8862e04a-a0a6-4dfa-b42e-2685b6dac9e5.png">





### Additional Notes



### Types of Changes


### Check all that apply
